### ISSUE TITLE
Add Fourier physics operators (Jacobian, inverse Laplacian, biharmonic, dealiasing)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#25]
## Overview

spectraldiffx has basic Fourier derivatives (∂/∂x, ∂²/∂x², gradient, laplacian) but lacks higher-level physics operators essential for QG/turbulence solvers.

## Missing Operators

| Operator | Formula | Use Case |
|----------|---------|----------|
| **Jacobian** | J(f,g) = ∂f/∂x·∂g/∂y − ∂f/∂y·∂g/∂x | QG advection — most critical gap |
| **Inverse Laplacian** | ∇⁻²f (Poisson solve in spectral space) | Streamfunction from vorticity |
| **Velocity from streamfunction** | u = −∂ψ/∂y, v = ∂ψ/∂x | QG/2D turbulence diagnostics |
| **Vorticity from velocity** | ζ = ∂v/∂x − ∂u/∂y | Curl in spectral space |
| **Biharmonic** | ∇⁴f | Scale-selective hyperdiffusion |
| **Hyperviscosity** | (−1)^{n+1} · ν · ∇^{2n} f | Parameterized-order dissipation |
| **Vector Laplacian** | ∇²u (component-wise) | Viscous term for velocity |
| **Dealiasing** | 2/3 rule mask generation + application | Quadratic nonlinearity handling |

## Reference Implementation

**Source:** `jej_vc_snippets/derivatives/spectral_fft_jax.py`

Implements all of the above as an equinox module with:
- Precomputed wavenumber arrays for 1D/2D/3D
- Dealiasing mask stored as module attribute
- All operators are JIT-compatible

## Proposed API

```python
class SpectralDerivative2D(eqx.Module):
    # Existing: derivative, gradient, laplacian
    
    # NEW:
    def jacobian(self, f_hat: Complex[Array, "Ny Nx"], g_hat: Complex[Array, "Ny Nx"]) -> Complex[Array, "Ny Nx"]:
        """J(f,g) = ∂f/∂x·∂g/∂y − ∂f/∂y·∂g/∂x (dealiased)"""
        
    def inverse_laplacian(self, f_hat: Complex[Array, "Ny Nx"]) -> Complex[Array, "Ny Nx"]:
        """∇⁻²f — Poisson solve in spectral space"""
    
    def velocity_from_streamfunction(self, psi_hat: Complex[Array, "Ny Nx"]) -> tuple:
        """u = −∂ψ/∂y, v = ∂ψ/∂x"""
    
    def vorticity_from_velocity(self, u_hat, v_hat) -> Complex[Array, "Ny Nx"]:
        """ζ = ∂v/∂x − ∂u/∂y"""
    
    def biharmonic(self, f_hat: Complex[Array, "Ny Nx"]) -> Complex[Array, "Ny Nx"]:
        """∇⁴f"""
    
    def hyperviscosity(self, f_hat, nu: float, order: int = 2) -> Complex[Array, "Ny Nx"]:
        """(−1)^{n+1} · ν · ∇^{2n} f"""
    
    def dealias(self, f_hat: Complex[Array, "Ny Nx"]) -> Complex[Array, "Ny Nx"]:
        """Apply 2/3 rule dealiasing mask"""
```

## Target Module

`spectraldiffx/_src/operators.py` (extend `SpectralDerivative1D/2D/3D`)

## Acceptance Criteria

- [ ] Jacobian operator with dealiasing (2D, 3D)
- [ ] Inverse Laplacian (Poisson solve in spectral space)
- [ ] Velocity ↔ streamfunction conversions
- [ ] Vorticity from velocity (spectral curl)
- [ ] Biharmonic operator (∇⁴)
- [ ] Hyperviscosity operator (arbitrary order ∇^{2n})
- [ ] Dealiasing utility (2/3 rule)
- [ ] Tests for all operators
- [ ] 1D, 2D, 3D variants where applicable

## Priority

**HIGH** — Jacobian and inverse Laplacian are essential for any spectral QG/turbulence solver. Blocks model zoo issues #10, #12, #13.
